Fix database export when using GRDB

bjtitus opened the pull request:
Fixes [PCIOS-85](https://linear.app/a8c/issue/PCIOS-85)

I think this is safe to include in the beta release because we still have some time and this _only_ affects the mostly debug export feature.

The `sqlite3-wal` file is also included:

> The WAL file exists for as long as any [database connection](https://www.sqlite.org/c3ref/sqlite3.html) has the database open. Usually, the WAL file is deleted automatically when the last connection to the database closes. However, if the last process to have the database open exits without cleanly shutting down the database connection, or if the [SQLITE_FCNTL_PERSIST_WAL](https://www.sqlite.org/c3ref/c_fcntl_begin_atomic_write.html#sqlitefcntlpersistwal) [file control](https://www.sqlite.org/c3ref/file_control.html) is used, then the WAL file might be retained on disk after all connections to the database have been closed. The WAL file is part of the persistent state of the database and should be kept with the database if the database is copied or moved.

via [sqlite.org](https://www.sqlite.org/wal.html)

## To test

* Profile > Help & Feedback > ⋯ > Export Database
* Delete the app
* Re-import that file by opening from Files or drag and dropping it to the simulator
* Import database file
* Ensure that Podcasts and Settings are re-imported

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Change the database export filename extension to sqlite3

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/98">7.95 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Add SQLite WAL database file to export/import

> The WAL file exists for as long as any [database connection](https://www.sqlite.org/c3ref/sqlite3.html) has the database open. Usually, the WAL file is deleted automatically when the last connection to the database closes. However, if the last process to have the database open exits without cleanly shutting down the database connection, or if the [SQLITE_FCNTL_PERSIST_WAL](https://www.sqlite.org/c3ref/c_fcntl_begin_atomic_write.html#sqlitefcntlpersistwal) [file control](https://www.sqlite.org/c3ref/file_control.html) is used, then the WAL file might be retained on disk after all connections to the database have been closed. The WAL file is part of the persistent state of the database and should be kept with the database if the database is copied or moved.

via [sqlite.org](https://www.sqlite.org/wal.html)

leandroalonso commented on podcasts/PCBundleDoc.swift:
```diff
@@ -10,7 +10,7 @@ struct PCBundleDoc: FileDocument {
 
     enum Constants {
-        static let databaseFilename = "database.sqlite"
+        static let databaseFilename = "database.sqlite3"
```
@bjtitus I think there are two things we need to take into account with GRDB: the `shm` and the `wal` files. Eg.:

<img width="174" height="69" alt="Image" src="https://github.com/user-attachments/assets/e7c8f302-587a-4c2a-8411-75288ea02a07" />

bjtitus replied:
Ok, yes. I was thinking these would be cleaned up with closing the database but it looks like that isn't always the case.

> The WAL file exists for as long as any [database connection](https://www.sqlite.org/c3ref/sqlite3.html) has the database open. Usually, the WAL file is deleted automatically when the last connection to the database closes. However, if the last process to have the database open exits without cleanly shutting down the database connection, or if the [SQLITE_FCNTL_PERSIST_WAL](https://www.sqlite.org/c3ref/c_fcntl_begin_atomic_write.html#sqlitefcntlpersistwal) [file control](https://www.sqlite.org/c3ref/file_control.html) is used, then the WAL file might be retained on disk after all connections to the database have been closed. The WAL file is part of the persistent state of the database and should be kept with the database if the database is copied or moved.

via [sqlite.org](https://www.sqlite.org/wal.html)

bjtitus replied:
@leandroalonso This is now updated with the wal file handling.

From what I can tell, the `shm` shared-memory file does not need to be included:
> The shared-memory file contains no persistent content. The only purpose of the shared-memory file is to provide a block of shared memory for use by multiple processes all accessing the same database in WAL mode. 

I also looked at the [backup method](https://github.com/groue/GRDB.swift#backup) provided by GRDB. I think that might be worth adding at some point but it seems to mostly work for now. What do you think?

danielebogo commented on podcasts/AppDelegate+UrlHandling.swift:
```diff
@@ -45,7 +45,11 @@ extension AppDelegate {
                     Task {
                         let fileWrapper = try FileWrapper(url: url)
-                        try PCBundleDoc.performImport(from: fileWrapper)
+                        do {
```
@bjtitus Would it make sense to include `let fileWrapper = try FileWrapper(url: url)` in `do { ... }` to catch also any possible error?

bjtitus replied:
Good point. Changed!

bjtitus committed:
Add FileWrapper to do catch

danielebogo reviewed (approved):
Tested as described and it worked
